codalab/apps/web/static/js/worksheet/worksheet_permission_toggle.jsx

```
@@ -0,0 +1,150 @@
+              type: 'GET',
+              contentType: 'application/json',
+            });
+
+            var second = first.then(function(data, textStatus, jqXHR) {
+              var worksheetData = {
+                  data: [{
+                      type: 'worksheet-permissions',
+                      attributes: {
+                        permission: newPermissionValue,
+                      },
+                      relationships: {
+                        worksheet: {
+                          data: {
+                            type: 'worksheets',
+                            id: self.props.ws.uuid,
+                          },
+                        },
+                        group: {
+                          data: {
+                            type: 'groups',
+                            id: data.data.id,
+                          },
+                        },
+                      },
+                  }]
+              };
+              return $.ajax({
+                  url: '/rest/worksheet-permissions',
+                  type: 'POST',
+                  contentType: 'application/json',
+                  data: JSON.stringify(worksheetData),
+              });
+            });
+            
+            var third = second.then(function (data, textStatus, jqXHR) {
+               // return index of public group, -1 if not present
+
+                var publicGroupIndex = getPublicGroupIndex(self.props.ws);
+                if (publicGroupIndex != -1) {
+                  // removing public access
+                  self.props.removePermission(publicGroupIndex);
+                } else {
+                  // adding public access
+                  self.props.addPermission(data.data[0].relationships.group.data.id, 'public', 1); // 1 is read permission
+                }
+
+                var bundleData = {data: []};
+                return $.ajax({
+                  url: '/rest/bundles?keywords=host_worksheet=' + self.props.ws.uuid,
```

As discussed below probably best to avoid paging (since it wouldn't be atomic) and just increase the max limit to the number of bundles on the worksheet as you said, and then let the request fail if it's too big.